File: Pythonleetcode/Leetcode108.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution(object):
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        def to_bst(nums,start,end):
            if start>end:
                return None
            mid=(end+start)//2
            # print(mid)
            node=TreeNode(nums[mid])
            node.left=to_bst(nums,start,mid-1)
            node.right = to_bst(nums, mid+1, end)

            return node

        return to_bst(nums,0,len(nums)-1)

# print(Solution().sortedArrayToBST(nums = [-10,-3,0,5,9]))

File: CppPractice/CppLeetcode/Leetcode108.cpp
class Solution {
public:
    vector<int> numsPublic;
    TreeNode* helper(int left,int right)
    {
        // cout<<left<<" "<<right<<endl;
        if(left>right){return nullptr;}
        int mid =left+(right-left)/2;
        TreeNode *newNode = new TreeNode(numsPublic[mid]);
        newNode->left = helper(left,mid-1);
        newNode->right = helper(mid+1,right);
        return newNode;

    }
    TreeNode* sortedArrayToBST(vector<int>& nums) 
    {
        numsPublic = nums;
        return helper(0,nums.size()-1);
    }
};